proto/proto/types/primitives.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+package primitives;
+
+// SMT
+// ================================================================================================
+
+// Represents a single SMT leaf entry.
+message SmtLeafEntry {
+    // The key of the entry.
+    Digest key = 1;
+
+    // The value of the entry.
+    Digest value = 2;
+}
+
+// Represents multiple leaf entries in an SMT.
+message SmtLeafEntryList {
+    // The entries list.
+    repeated SmtLeafEntry entries = 1;
+}
+
+// A leaf in an SMT, sitting at depth 64. A leaf can contain 0, 1 or multiple leaf entries.
+message SmtLeaf {
+    oneof leaf {
+        // An empty leaf.
+        uint64 empty = 1;
```

Comment:
This could probably be the empty type `google.proto.Empty` (or whatever it actually is).

Reply:
I think that we still want the leaf index (that's what the `empty` represents here). Probably it is better to just rename it